### DISCUSSION TITLE
how to parse C style multi line comment in verison 1.0

### DISCUSSION THREAD

**Discussion by myrfy001** [zesterer/chumsky#571] [Q&A] [answered]
I have a working version written in version 0.9 as following:
```
fn parser_comment() -> impl Parser<char, (Token, Span), Error = Simple<char>> {
    let t1 = just("//")
        .ignore_then(take_until(just('\n').ignored()))
        .padded()
        .map_with_span(|(c, _), span| {
            let s = c.iter().collect();
            (Token::Comment(s), span)
        });

    let t2 = just("/*")
        .ignore_then(take_until(just("*/").ignored()))
        .padded()
        .map_with_span(|(c, _), span| {
            let s = c.iter().collect();
            (Token::Comment(s), span)
        });
    t1.or(t2)
}
```

Since take_until is removed in 1.0, how can I change it to work with 1.0?

I tried the following code: 
```
fn parser_comment<'src>(
) -> impl Parser<'src, LexerInput<'src>, Spanned<Token<'src>>, extra::Err<Rich<'src, char, Span>>> + Clone
{
    let t1 = just("//")
        .then(none_of("\n").repeated().to_slice())
        .to_slice()
        .boxed();

    let t2 = just("/*")
        .then(none_of(just("*/")).repeated().to_slice())
        .to_slice()
        .boxed();
    t1.or(t2).map_with(|c, e| (Token::Comment(c), e.span()))
}
```

But putting a `just` inside `none_of` seems not work

**Comment by myrfy001** [comment, marked as answer]
I have solved this with `and_is`  + `not`.

By the way, is there any document that tells how to migrate from v0.x to v1.0? Or tell the API difference between them?

**Reply by zesterer** [reply]
Good to hear. Yes, `and_is` and `not` are the best replacements here. They're a strictly more powerful pair than `take_until` and can be used in many more cases.

There isn't a migration guide yet, but I'm planning to write one before 1.0 becomes stable. Of course, you're very welcome to open issues/discussions here for anything you get stuck on :)
